superset/databases/api.py

```diff
@@ -1341,6 +1342,7 @@ def validate_sql(self, pk: int) -> FlaskResponse:
             return self.response_404()
 
     @expose("/oauth2/", methods=["GET"])
+    @transaction()
```

Comment:
Nice!

Does this work even though the method is a `GET`? I remember you mentioning that it was not the case?